src/testing/flow_solver_cases/naca0012.cpp

```diff
@@ -0,0 +1,95 @@
+#include "naca0012.h"
+#include <deal.II/base/function.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_refinement.h>
+#include <deal.II/grid/grid_tools.h>
+#include <stdlib.h>
+#include <iostream>
+#include <deal.II/dofs/dof_tools.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/fe/fe_values.h>
+#include "physics/physics_factory.h"
+#include "dg/dg.h"
+#include <deal.II/base/table_handler.h>
+#include "mesh/grids/naca_airfoil_grid.hpp"
+#include "mesh/gmsh_reader.hpp"
+#include "functional/lift_drag.hpp"
+
+namespace PHiLiP {
+namespace Tests {
+//=========================================================
+// NACA0012
+//=========================================================
+template <int dim, int nstate>
+NACA0012<dim, nstate>::NACA0012(const PHiLiP::Parameters::AllParameters *const parameters_input)
+        : FlowSolverCaseBase<dim, nstate>(parameters_input)
+{
+}
+
+template <int dim, int nstate>
+void NACA0012<dim,nstate>::display_flow_solver_setup() const
+{
+    using PDE_enum = Parameters::AllParameters::PartialDifferentialEquation;
+    const PDE_enum pde_type = this->all_param.pde_type;
+    const double pi = atan(1.0) * 4.0;
+    std::string pde_string;
+    if (pde_type == PDE_enum::euler)                {pde_string = "euler";}
+    if (pde_type == PDE_enum::navier_stokes)        {pde_string = "navier_stokes";}
+    this->pcout << "- PDE Type: " << pde_string << std::endl;
+    this->pcout << "- Polynomial degree: " << this->all_param.grid_refinement_study_param.poly_degree << std::endl;
+    this->pcout << "- Courant-Friedrich-Lewy number: " << this->all_param.flow_solver_param.courant_friedrich_lewy_number << std::endl;
+    this->pcout << "- Freestream Reynolds number: " << this->all_param.navier_stokes_param.reynolds_number_inf << std::endl;
+    this->pcout << "- Freestream Mach number: " << this->all_param.euler_param.mach_inf << std::endl;
+    this->pcout << "- Angle of attack: " << this->all_param.euler_param.angle_of_attack*180/pi << std::endl;
+    this->pcout << "- Side-slip angle: " << this->all_param.euler_param.side_slip_angle*180/pi << std::endl;
+}
+
+template <int dim, int nstate>
+std::shared_ptr<Triangulation> NACA0012<dim,nstate>::generate_grid() const
+{
+    //Dummy triangulation
+    std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation>(MPI_COMM_WORLD);
+    dealii::GridGenerator::Airfoil::AdditionalData airfoil_data;
+    dealii::GridGenerator::Airfoil::create_triangulation(*grid, airfoil_data);
+    grid->refine_global();
+    return grid;
```

Comment:
Also, can this be moved to `src/mesh/grids/` and then you call it here instead? We want the `FlowSolverCaseBase` classes to simply call the appropriate grid from the files in that directory; see `src/flow_solver_cases/periodic_cube_flow.cpp`. Ideally we'd have a grid factory where we just pass the parameters and it returns the grid, that way we'd have only one `generate_grid` definition that would be in the base class. There seems to be a file `src/mesh/grids/naca_airfoil_grip.cpp`, can we use that? @dougshidong

Reply:
This grid isn't actually used, its just a dummy triangulation because dg needs it to be initialized. Doug opened issue #135 for this. I was also having some issues yesterday with the naca_airfoil function in naca_airfoil_grid.cpp, you can see one of the conversations with Doug about it. We probably shouldn't bother since its just a dummy triangulation anyway.